### PR TITLE
PWA-1906 upward php inlining

### PR DESCRIPTION
**Pull request by justinconabree**
## Description
Adding new resolver type of "Computed" which uses a mapped class to resolve the value. This keeps business logic out of UPWARD directly, but allows for high-performance data inlining.

Related story: https://jira.corp.magento.com/browse/PWA-1906

### CODE REVIEW

File: src/Resolver/Computed.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?php
+
```
Comment (larsroettig) [inline src/Resolver/Computed.php:6]: copwrite

File: src/Resolver/Computed.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+use Magento\UpwardConnector\Model\Computed\ComputedInterface;
+
+class Computed extends AbstractResolver
+{
+    /** @var \Magento\UpwardConnector\Model\ComputedPool */
+    private $computed;
+
+    public function __construct()
+    {
+        $this->computed = \Magento\Framework\App\ObjectManager::getInstance()
```
Comment (larsroettig) [inline src/Resolver/Computed.php:21]: Should be possible to create this via dependency injection `ObjectManage` is only allowed for Factory and also not in constructor

Comment (justinconabree) [reply src/Resolver/Computed.php:21]: @larsroettig Unfortunately they instantiate the class directly with `new self::$resolverClasses[$resolverType]()` so dependency injection won't work. This is essentially a way to go from UPWARD (no dependency injection) back into Magento coding practices (with dependency injection).

**Comment by justinconabree** [comment]
Will open new PR for PWA-1204 to avoid overlapping code review

**Pull request closed without merging**